packages/schema-generator/src/generate_schema.ts

```diff
@@ -0,0 +1,117 @@
+import { TypescriptDataSchemaGenerator } from '@aws-amplify/graphql-schema-generator';
+import fs from 'fs/promises';
+import { AmplifyUserError } from '@aws-amplify/platform-core';
+
+export type SchemaGeneratorConfig = {
+  connectionString: string;
+  out: string;
+};
+
+/**
+ * Schema generator class.
+ */
+export class SchemaGenerator {
+  generate = async (props: SchemaGeneratorConfig) => {
+    const dbConfig = parseDatabaseUrl(props.connectionString);
+
+    try {
+      const schema = await TypescriptDataSchemaGenerator.generate(dbConfig);
+      await fs.writeFile(props.out, schema);
+    } catch (err) {
+      const databaseError = err as DatabaseConnectError;
+      if (databaseError.code === 'ETIMEDOUT') {
+        throw new AmplifyUserError(
+          'DatabaseConnectionError',
+          {
+            message: `Unable to connect to the database at ${dbConfig.host}:${dbConfig.port}. `,
+            resolution:
+              'Check if the credentials are correct. Also, check if the database is running and accessible from the network.',
+          },
+          databaseError
+        );
+      }
+      throw err;
+    }
+  };
+}
+
+// eslint-disable-next-line spellcheck/spell-checker
```

Comment:
can you remove suppressions after adding to dictionary ?

Reply:
Done.